source/statements.tex

```diff
@@ -146,6 +146,18 @@
 \indextext{label!\idxcode{default}}%
 Case labels and default labels shall occur only in \keyword{switch} statements.
 
+\pnum
+A \defnadj{control-flow-limited}{statement} is a statement \tcode{S} for which:
+\begin{itemize}
+\item
+  a \keyword{case} or \keyword{default} label appearing within \tcode{S} shall
+  be associated with a \keyword{switch} statement\iref{stmt.switch} within
+  \tcode{S}, and
+\item
+  a label declared in \tcode{S} shall only be referred to by a
+  statement\iref{stmt.goto} in \tcode{S}.
```

Comment:
I realize this is original wording, but should we narrow this down to "by a _goto_ statement", not just any old statement? Which other kinds of statement could be referring to a label?

Reply:
gcc-style computed gotos are similarly constrained, I believe, so keeping the "goto" a bit low-profile helps that extension. This is not a strong argument.

Reply:
Aha, I was wondering if that was behind it! That's fine, esp. since this is pre-existing. It just looked a bit odd to me to say "statement" but then cross-reference stmt.goto.

Fine to ship!

Reply:
I think this just falls out of the existing wording and was not by design of the wording. But we might not want to move towards the other direction, now that we've discovered what the wording does.

Reply:
By "the other direction" you mean narrowing it down to just `goto` statements? So you'd just keep it as is?

Reply:
Yes, I'd keep it as-is.

Reply:
Off we go.